sdk/lib/udmihelp/udmihelp.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * PROJECT:     ReactOS User-mode DMI/SMBIOS Helper Functions
+ * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:     SMBIOS table parsing functions
+ * COPYRIGHT:   Copyright 2018 Stanislav Motylkov
+ */
+
+#include "precomp.h"
+
+static UINT (WINAPI * pGetSystemFirmwareTable)(DWORD, DWORD, PVOID, DWORD);
+
+/* Load SMBIOS Data */
+PVOID
+LoadSMBiosData(
+    _Inout_updates_(ID_STRINGS_MAX) PCHAR * Strings)
+{
+    HANDLE hKernel;
+    PVOID pBuffer = NULL;
+    HKEY hKey;
+    DWORD dwType, dwCheck, dwBytes = 0;
+
+    hKernel = GetModuleHandleW(L"kernel32.dll");
```

Comment:
To improve this (ie. not to do that everytime), I would suggest something like this:
1. Having a global static BOOLean that indicates whether we have already tried to load kernel32.dll and find the API in it;
2. Here in this function, check whether this boolean is FALSE. If it's FALSE, then we find the handle to kernel32 and try to initialize pGetSystemFirmwareTable (may fail if the API doesn't exist). Then the boolean is set to TRUE regardless of the success.
3. So that next time the same program calls again LoadSMBiosData(), we re-use pGetSystemFirmwareTable without tring to find it again in kernel32, because the boolean is now TRUE.
Note that I don't just check whether pGetSystemFirmwareTable is NULL or not, because it can be NULL just because the API was not found in kernel32 and therefore it's also not needed to try to reload and re-find it again (and again).